packages/graphql/src/translate/where/create-where-and-params.ts

```diff
@@ -0,0 +1,264 @@
+            if (!refNode) throw new Error("Relationship filters must reference nodes");
+            const labels = refNode.getLabelString(context);
+
+            const inStr = relationField.direction === "IN" ? "<-" : "-";
+            const outStr = relationField.direction === "OUT" ? "->" : "-";
+            const relTypeStr = `[:${relationField.type}]`;
+
+            if (value === null) {
+                res.clauses.push(`${isNot ? "" : "NOT "}EXISTS((${varName})${inStr}${relTypeStr}${outStr}(${labels}))`);
+                return res;
+            }
+
+            let resultStr = [
+                `EXISTS((${varName})${inStr}${relTypeStr}${outStr}(${labels}))`,
+                `AND ${listPredicate}(${param} IN [(${varName})${inStr}${relTypeStr}${outStr}(${param}${labels}) | ${param}] INNER_WHERE `,
+            ].join(" ");
+
+            const recurse = createWhereAndParams({
+                whereInput: value,
+                varName: param,
+                chainStr: param,
+                node: refNode,
+                context,
+                recursing: true,
+            });
+
+            if (recurse[0]) {
+                resultStr += recurse[0];
+                resultStr += ")"; // close predicate
+                res.clauses.push(resultStr);
+                res.params = { ...res.params, ...recurse[1] };
+            }
+
+            return res;
+        }
+
+        // Connection Field
+
+        const connectionField = node.connectionFields.find((x) => x.fieldName === fieldName);
+
+        if (connectionField) {
+            let nodeEntries: Record<string, any> = value;
+
+            if (!connectionField?.relationship.union) {
+                nodeEntries = { [connectionField.relationship.typeMeta.name]: value };
+            }
+
+            Object.entries(nodeEntries).forEach((entry) => {
+                const refNode = context.neoSchema.nodes.find((x) => x.name === entry[0]) as Node;
+                const relationship = context.neoSchema.relationships.find(
+                    (x) => x.name === connectionField.relationshipTypeName
+                ) as Relationship;
+
+                const thisParam = `${param}_${refNode.name}`;
+                const relationshipVariable = `${thisParam}_${connectionField.relationshipTypeName}`;
+                const inStr = connectionField.relationship.direction === "IN" ? "<-" : "-";
+                const outStr = connectionField.relationship.direction === "OUT" ? "->" : "-";
+                const labels = refNode.getLabelString(context);
+                const collectedMap = `${thisParam}_map`;
+
+                if (value === null) {
+                    res.clauses.push(
+                        `${isNot ? "" : "NOT "}EXISTS((${varName})${inStr}[:${
+                            connectionField.relationship.type
+                        }]${outStr}(${labels}))`
+                    );
+                    return res;
```

Comment:
Nice. Good catch. The other ESLint issue is the dependency cycle introduced in a previous commit.

Reply:
For sure - we're not 100% enforcing ESLint just yet because as you say, there are a lot of legacy violations getting in the way at the moment. Thanks!